spring-cloud-task-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -153,3 +153,184 @@ public class TimestampTaskApplication {
+    ...
+</configuration
+----
+
+If you have an artifact that is only available through a private internal maven repository (may be an enterprise wide
+Nexus repo that you use globally across teams), and you need that for your app, you can define that as part of the maven
+plugin configuration.
+
+For example,
+
+[source,xml]
+----
+<configuration>
+...
+    <extraRepositories>
+        <repository>
+            <id>private-internal-nexus</id>
+            <url>.../</url>
+            <name>...</name>
+            <snapshotEnabled>...</snapshotEnabled>
+        </repository>
+    </extraRepositories>
+</configuration>
+----
+
+Then you can define this as part of your app tag:
+
+[source,xml]
+----
+<foobar-task>
+    <extraRepositories>
+        <private-internal-nexus />
+    </extraRepositories>
+</foobar-task>
+----
+
+* cd into the directory where you generated the apps (`apps` at the root of the repository by default, unless you changed
+it elsewhere as described above).
+
+Here you will see `foobar-task` along with all the other out of the box apps that is generated.
+
+If you only care about the foobar-task apps and nothing else, you can cd into that directory
+and import it directly into your IDE of choice. Each of them is a self contained spring boot application project.
+For all the generated apps, the parent is `spring-boot-starter-parent`. There is no way to change that around as ultimately these
+are spring boot based applications and that parent would bring many necessary spring components into your application.
+
+You can cd into these custom foobar-task directories and do the following to build the apps:
+
+`cd foobar-task`
+
+`mvn clean install`
+
+This would install the foobar-task into your local maven cache (~/.m2 by default).
+
+The app generation phase above would add an integration test to the app project that is making sure that all the spring
+components and contexts are loaded properly. However, these tests are not run by default when you do a `mvn install`.
+You can force the running of these tests by doing the following:
+
+`mvn clean install -DskipTests=false`
+
+* Now that you built the applications, they are available under the `target` directories of the respective apps and also as
+maven artifacts in your local maven repository. Go to the `target` directory and run the following:
+
+`java -jar foobar-task.jar`
+
+It should start the application up.
+
+* The generated apps also support the creation of docker images. You can cd into one of the foobar-task app and do the
+following:
+
+`mvn clean package docker:build`
+
+This would create the docker image under the `target/docker/springcloudtask` directory. Please ensure that the Docker
+container is up and running and DOCKER_HOST environment variable is properly set before you try `docker:build`.
+
+All the generated apps from the repository are uploaded to the docker hub at: https://hub.docker.com/u/springcloudtask/
```

Comment:
If someone wants to use the app starters infrastructure to create a new starter/app, and then host somewhere on docherhub other than springcloudtask repo, they can follow this step to do that.